lib/std/fmt.zig

```diff
@@ -464,6 +464,25 @@ pub fn formatType(
             });
             return formatType(@as(Slice, &value), fmt, options, context, Errors, output, max_depth);
         },
+        .Vector => {
+            if (comptime std.meta.trait.hasFn("format")(T)) {
+                return value.format(fmt, options, context, Errors, output);
+            }
+
+            if (max_depth == 0) {
```

Comment:
Printing vectors isn't recursive so this isn't needed.

Reply:
I think it's needed if struct/array has vectors.

Reply:
That’s only to prevent potentially infinite recursion. Vectors are scalars and you can’t blow out the stack with them, so it’s a lot more like enums or arrays.

Reply:
Done.